rest-api/tools/setup_database.sh

```diff
@@ -100,36 +91,50 @@ function finish {
 }
 trap finish EXIT
 
-echo '{"db_connection_string": "'$CONNECTION_STRING'", ' \
-     ' "backup_db_connection_string": "'$BACKUP_CONNECTION_STRING'", '\
-     ' "rdr_db_password": "'$RDR_PASSWORD'", ' \
-     ' "root_db_password": "'$ROOT_PASSWORD'", ' \
-     ' "read_only_db_password": "'$READONLY_PASSWORD'", ' \
-     ' "db_connection_name": "'$INSTANCE_CONNECTION_NAME'", '\
-     ' "backup_db_connection_name": "'$BACKUP_INSTANCE_NAME'", '\
-     ' "db_user": "'$RDR_DB_USER'", '\
-     ' "db_name": "'$DB_NAME'" }' > $TMP_DB_INFO_FILE
+function get_existing_db_config {
+  "Getting database info..."
+  tools/install_config.sh --key db_config --instance $INSTANCE \
+   --creds_file ${CREDS_FILE} --config_output "$TMP_DB_INFO_FILE"
+}
 
 for db_name in "rdr" "metrics"; do
   if [ "${UPDATE_PASSWORDS}" = "Y" ]
   then
-    cat tools/update_passwords.sql | envsubst > $UPDATE_DB_FILE
+    	echo "Updating database user passwords..."
+	randpw
```

Comment:
So create_db.sql does reference these passwords:

https://github.com/all-of-us/raw-data-repository/blob/master/rest-api/tools/create_db.sql#L3

It just doesn't use them if the database already exists.

So I think this password generation stuff needs to run at least in the DB creation case, if not the update the existing DB case.

Reply:
done